### PR TITLE
volume: Don't decrement refcount below 0 

### PR DESCRIPTION
**Pull request by vvoland**
- related to: https://github.com/moby/moby/issues/46508

With both rootless and live restore enabled, there's some race condition which causes the container to be `Unmount`ed before the refcount is restored.

This makes sure we don't underflow the refcount (uint64) when decrementing it.

The root cause of this race condition still needs to be investigated and fixed, but at least this unflakies the `TestLiveRestore`.

**- How to verify it**
Check `TestLiveRestore`
The CI is green after running the test 30x times: https://github.com/moby/moby/pull/47530/commits/afa01acc55fe383ac32325bdaa8bfc039b9c1b50

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by vvoland** [comment]
The CI is green after running the test 30x times: https://github.com/moby/moby/pull/47530/commits/afa01acc55fe383ac32325bdaa8bfc039b9c1b50
Will remove the test commit now.

File: volume/local/local.go [hunk 1]
```diff
@@ -334,8 +335,14 @@ func (v *localVolume) Unmount(id string) error {
 	// ultimately there's nothing that can be done. If we don't decrement the count
 	// this volume can never be removed until a daemon restart occurs.
 	if v.needsMount() {
-		v.active.count--
-		logger.WithField("active mounts", v.active).Debug("Decremented active mount count")
+		if v.active.count > 0 {
```
Comment (thaJeztah) [inline volume/local/local.go:338]: can you add a TODO / FIXME?

Comment (vvoland) [reply volume/local/local.go:339]: Done

**Review by thaJeztah — APPROVED** [review]
LGTM

**Pull request merged**